### DISCUSSION TITLE
Fastlane with App Store Connection Key wants 2FA session cookies

### DISCUSSION THREAD

**Discussion by felstell** [fastlane/fastlane#19143] [Q&A]
Hello there! 

I tried to change auth method in my Fastlane from Spaceuath to App Store Connect Key. My fasfile looks like 

```
app_store_connect_api_key(
    key_id: "***********",
    issuer_id: "*************************",
    key_filepath: "../../../KeyFileName.p8",
    duration: 1200,
    in_house: false
)
pilot()
```

But always when I tried to call this (or with App Specific Password) lane from TeamCity I have got this message: 

![CleanShot 2021-07-23 at 15 15 54@2x](https://user-images.githubusercontent.com/8091787/126780501-ec5740ac-e3ff-4240-9be7-a2475c32e677.png)

What is the problem here?

**Comment by crazymanish** [comment]
Hello @felstell 

Can you try below, please

```ruby
api_key = app_store_connect_api_key(
    key_id: "***********",
    issuer_id: "*************************",
    key_filepath: "../../../KeyFileName.p8",
    duration: 1200,
    in_house: false
)
pilot(api_key: api_key)
